ogr/ogrlibjsonutils.cpp

```diff
@@ -132,22 +133,28 @@ static int OGR_json_double_with_precision_to_string(struct json_object *jso,
         json_object_get_userdata(jso);
 #endif
     // Precision is stored as a uintptr_t content casted to void*
-    const uintptr_t nPrecision = reinterpret_cast<uintptr_t>(userData);
-    char szBuffer[75] = {};
+    const uintptr_t nPrecisionIn = reinterpret_cast<uintptr_t>(userData);
     const double dfVal = json_object_get_double(jso);
     if (fabs(dfVal) > 1e50 && !std::isinf(dfVal))
     {
+        char szBuffer[75] = {};
         CPLsnprintf(szBuffer, sizeof(szBuffer), "%.17g", dfVal);
+        return printbuf_memappend(pb, szBuffer,
```

Comment:
Can you use the return value of `CPLsnprintf` here instead of `strlen`? It should fit the buffer.

Reply:
that's not the code path taken here, but indeed good suggestion. Fixed

Reply:
Right, I couldn't look at `OGRFormatDouble`.